### PR TITLE
[dynamo] Some torchrec_dlrm related fixes

### PR DESCRIPTION
**Pull request by anijain2305**
Issue 1 of https://github.com/pytorch/pytorch/issues/101918

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

### CODE REVIEW

File: torch/_dynamo/output_graph.py [hunk 1]
```diff
@@ -609,14 +609,10 @@ def register_leaf_name(leaf_name):
                     # annoying, but there are cases when we do not have parameters
                     # see test_nn_moduledict_contains
                     if hasattr(target, "_parameters"):
-                        for leaf_name, _ in target.named_parameters(
-                            remove_duplicate=False
-                        ):
+                        for leaf_name, _ in target.named_parameters():
```
Comment (anijain2305) [inline torch/_dynamo/output_graph.py:612]: @voznesenskym Is there any reason for remove_duplicate to be `False`?

Comment (voznesenskym) [reply torch/_dynamo/output_graph.py:612]: IIRC If you remove duplicates, you will get incomplete leaf names for AOTAutograd, and therefore name mismatches.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**